cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -185,127 +186,229 @@ func (pfct preflightCheckTest) Check() (warning, errorList []error) {
 		},
 	}
+	if _, err = f.WriteString("Test FileContentCheck"); err != nil {
+		t.Fatalf("Failed to write to file: %v", err)
+	}
 	for _, rt := range tests {
-		// TODO: Make RunInitNodeChecks accept a ClusterConfiguration object instead of InitConfiguration
-		actual := RunInitNodeChecks(exec.New(), rt.cfg, sets.NewString(), rt.isSecondaryControlPlane, rt.downloadCerts)
-		if (actual == nil) != rt.expected {
+		_, output := rt.check.Check()
+		if (len(output) == 0) != rt.expectedError {
 			t.Errorf(
-				"failed RunInitNodeChecks:%v\n\texpected: %t\n\t  actual: %t\n\t error: %v",
+				"Failed FileContentCheck:%v\n\texpectedError: %t\n\t  actual: %t",
 				rt.name,
-				rt.expected,
-				(actual == nil),
-				actual,
+				rt.expectedError,
+				(len(output) == 0),
 			)
 		}
 	}
 }
 
-func TestRunJoinNodeChecks(t *testing.T) {
+func TestDirAvailableCheck(t *testing.T) {
+	fileDir, err := ioutil.TempDir("/tmp", "dir-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(fileDir)
 	var tests = []struct {
-		name     string
-		cfg      *kubeadmapi.JoinConfiguration
-		expected bool
+		name          string
+		check         DirAvailableCheck
+		expectedError bool
 	}{
 		{
-			name:     "Check empty JoinConfiguration",
-			cfg:      &kubeadmapi.JoinConfiguration{},
-			expected: false,
+			name: "Directory exists and is empty",
+			check: DirAvailableCheck{
+				Path: fileDir,
+			},
+			expectedError: true,
 		},
 		{
-			name: "Check TLS Bootstrap APIServerEndpoint IPv4 addr",
-			cfg: &kubeadmapi.JoinConfiguration{
-				Discovery: kubeadmapi.Discovery{
-					BootstrapToken: &kubeadmapi.BootstrapTokenDiscovery{
-						APIServerEndpoint: "192.168.1.15",
-					},
-				},
+			name: "Directory exists and has something",
+			check: DirAvailableCheck{
+				Path: "/tmp",
 			},
-			expected: false,
+			expectedError: false,
 		},
 		{
-			name: "Check TLS Bootstrap APIServerEndpoint IPv6 addr",
-			cfg: &kubeadmapi.JoinConfiguration{
-				Discovery: kubeadmapi.Discovery{
-					BootstrapToken: &kubeadmapi.BootstrapTokenDiscovery{
-						APIServerEndpoint: "2001:1234::1:15",
-					},
-				},
+			name: "Directory does not exist",
+			check: DirAvailableCheck{
+				Path: "/does/not/exist",
 			},
-			expected: false,
+			expectedError: true,
 		},
 	}
+	for _, rt := range tests {
+		_, output := rt.check.Check()
+		if (output == nil) != rt.expectedError {
+			t.Errorf(
+				"Failed DirAvailableCheck:%v\n\texpectedError: %t\n\t  actual: %t",
+				rt.name,
+				rt.expectedError,
+				(output != nil),
+			)
+		}
+	}
+}
 
+func TestPortOpenCheck(t *testing.T) {
+	ln, _ := net.Listen("tcp", fmt.Sprintf("127.0.0.1:1000"))
+	defer ln.Close()
+	var tests = []struct {
+		name          string
+		check         PortOpenCheck
+		expectedError bool
+	}{
+		{
+			name:          "Port is available",
+			check:         PortOpenCheck{port: 0},
+			expectedError: true,
+		},
+		{
+			name:          "Port is not available",
+			check:         PortOpenCheck{port: 1000},
```

Comment:
```suggestion
			check:         PortOpenCheck{port: ln.Addr().(*net.TCPAddr).Port)},
```

see https://stackoverflow.com/a/43425461